src/memory.rs

```diff
@@ -50,6 +51,25 @@ impl Pointer {
             offset: 0,
         }
     }
+    pub fn is_aligned_to(&self, align: usize) -> bool {
```

Comment:
I'm wondering if this should take the `self.align` into account (i.e. the alignment of the base of the allocation). For example, if some pointer `p` was 2-aligned and at offset 0, then `p.is_aligned_to(4)` returns true, but it wouldn't necessarily be true in reality. A 2-aligned pointer *could* be 4-aligned, but it could also not be.